### PR TITLE
Shopping cart

### PR DESCRIPTION
**Pull request by wulalasuperdope**
- added cart api, cartThunks
- set up ShoppingCart and CarItem component
- fixed styling/spacing inconsistency in many files ;D

### CODE REVIEW

File: app/components/CartItem.js [hunk 1]
```diff
@@ -1,7 +1,36 @@
...
+import Product from "./Product";
 
-const CartItem = () => {
-  return <div>This is a cart item</div>;
-};
+class CartItem extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      productImage: this.props.productImage,
```
Comment (ShelOrock) [inline app/components/CartItem.js]: i would get this info from the redux store.

File: app/redux/actions.js [hunk 2]
```diff
@@ -1,64 +1,88 @@
...
+};
+
+export const setCart = products => {
+  return {
+    type: SET_CART,
+    products
+  };
+};
+
+export const _addItemToCart = product => {
```
Comment (ShelOrock) [inline app/redux/actions.js]: i think we just need to set the cart. our thunks will call a post or delete and then call another thunk to re-get which will set cart

File: app/redux/thunks/CartThunks.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
+import axios from "axios";
+
+import { setCart, addItemToCart, _removeItemFromCart } from "../actions";
+// temporary fetchCart thunk
+export function fetchCart() {
```
Comment (ShelOrock) [inline app/redux/thunks/CartThunks.js]: pass a userId into the function. the route will call `/api/users/${ :userId }/cart`

File: app/redux/thunks/CartThunks.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  return function thunk(dispatch) {
+    return axios
+      .get(`/api/cart`)
+      .then(res => res.data)
+      .then(products => dispatch(setCart(products)))
+      .catch(e => console.error("Error fetching Cart", e));
```
Comment (ShelOrock) [inline app/redux/thunks/CartThunks.js]: use my toast. dispatch(statusMessage({ status: FAIL, text: ' message goes here' })

File: app/redux/thunks/CartThunks.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  return function thunk(dispatch) {
+    return axios
+      .get(`/api/cart/${userId}`)
+      .then(res => res.data)
+      .then(products => dispatch(setCart(products)))
+      .catch(e => console.error("Error fetching Cart", e));
+  };
+}
+
+export function addItemToCart() {
```
Comment (ShelOrock) [inline app/redux/thunks/CartThunks.js]: pass a userId and a product. api call will be `/api/user/${ userId }/cart`

File: app/redux/thunks/CartThunks.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+  return function thunk(dispatch) {
+    return axios
+      .post(`/api/cart/add/${userId}`, product)
+      .then(res => res.data)
+      .then(product => dispatch(_addItemToCart(product)))
+      .catch(e => console.error("Error adding Cart item", e));
+  };
+}
+
+export function removeItemFromCart(product) {
```
Comment (ShelOrock) [inline app/redux/thunks/CartThunks.js]: pass a pass a userId and a product. api call will be /api/user/${ userId }/cart/${ productId }

File: app/redux/reducers.js [hunk 7]
```diff
@@ -84,3 +87,16 @@ export const wishlist = (state = [], action) => {
       return state;
   }
 };
+
```
Comment (ShelOrock) [inline app/redux/reducers.js:99]: consider two thunk calls over having a complicated reducer.

File: seed.js [hunk 8]
```diff
@@ -1,5 +1,5 @@
-const { users, products } = require('./seed-data.js');
-const { User, Product } = require('./server/db/models/index.js');
+const { users, products, carts } = require("./seed-data.js");
+const { User, Product, Cart } = require("./server/db/models/index.js");
 
 const seedUsers = () => {
```
Comment (ShelOrock) [inline seed.js]: make cart seed after user successfully seeds.

File: server/api/cart.js [hunk 9]
```diff
@@ -0,0 +1,49 @@
+const router = require("express").Router();
+const { models } = require("../db/index.js");
+const { Cart, Product } = models;
+//temporary get route
+router.get("/", (req, res, next) => {
```
Comment (ShelOrock) [inline server/api/cart.js]: move these into the user routes. so we can get a cart based on user on /api/user/:userId/cart

File: server/api/cart.js [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+    where: {
+      userId: req.params.userId
+    },
+    include: [{ model: Product }]
+  })
+    .then(productsInCart => res.status(200).send(productsInCart))
+    .catch(e => res.status(400).next(e));
+});
+*/
+router.post("/add/:userId", (req, res, next) => {
```
Comment (ShelOrock) [inline server/api/cart.js]: post request just on /api/user/:userId/cart

File: server/api/cart.js [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+  Cart.create(req.body)
+    .then(product => res.status(201).send(product))
+    .catch(e => res.status(400).next(e));
+});
+
+router.put("/edit/:productId", (req, res, next) => {
```
Comment (ShelOrock) [inline server/api/cart.js]: see above

File: server/api/cart.js [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+    {
+      where: { productid: req.params.productId },
+      returning: true
+    }
+  )
+    .then(() => res.status(202))
+    .catch(e => res.status(304).next(e));
+});
+
+router.delete("/remove/:productId", (req, res, next) => {
```
Comment (ShelOrock) [inline server/api/cart.js]: see above

File: server/api/users.js [hunk 13]
```diff
@@ -52,23 +52,20 @@ router.post('/', (req, res, next) => {
     billingState: billingState || null,
     billingZip: billingZip || null
   })
-    .then(user =>
```
Comment (ShelOrock) [inline server/api/users.js]: inconsistent linting.

**Review by ShelOrock — COMMENTED** [review]
See comments. over all good. Biggest priorities - api routes. and api calls.

**Comment by ShelOrock** [comment]
We need to make sure we're on the same page with linting (i.e. double quotes vs single quotes, tabs vs spaces, indentation, etc)

**Comment by wulalasuperdope** [comment]
Cart only works with seeded cart database for now. Will need to incorporate "Add to Cart" feature.

**Comment by wulalasuperdope** [comment]
Changes this PR:
- added api routes to users for fetching cart items, editing item quantity and removing item from cart
- added cart thunks for fetching cart items and removing item
- added content to ShoppingCart and CartItem components


Acceptance Criteria:
- Pull this branch
- Run the app
- Go to Cart page (you will see seeded cart database)
- Edit cart item quantity
- Remove car item

**Review by nries1 — DISMISSED** [review]
Nice job! I wasn't able to edit the quantity but removing cart items worked and the items rendered nicely. Thanks! You're good to merge.

**Review by nries1 — APPROVED** [review]
Nice job! I wasn't able to edit the quantity but removing cart items worked and the items rendered nicely. Thanks! You're good to merge.

**Pull request merged**